src/modalities/models/vision_transformer/vision_transformer_model.py

```diff
@@ -0,0 +1,170 @@
+        x = x + self.attention(self.norm1(x))
+        x = x + self.mlp(self.norm2(x))
+        return x
+
+
+class VisionTransformer(nn.Module):
+    """ViT
+
+    The Vision Transformer (ViT) is a pure transformer architecture
+    that applies attention mechanisms directly to sequences of image patches for image classification tasks.
+
+    Paper: `An Image is Worth 16x16 Words: Transformers for Image Recognition at Scale`
+    Link: https://arxiv.org/abs/2010.11929
+    """
+
+    def __init__(
+        self,
+        sample_key: str,
+        prediction_key: str,
+        img_size: Union[Tuple[int, int], int] = 224,
+        n_classes: int = 1000,
+        n_layer: int = 12,
+        attention_config: AttentionConfig = None,
+        n_head: int = 8,
+        n_embd: int = 768,
+        ffn_hidden: int = 3072,
+        dropout: float = 0.0,
+        patch_size: int = 16,
+        patch_stride: int = 16,
+        n_img_channels: int = 3,
+        add_cls_token: bool = True,
+        bias: bool = True,
+    ) -> None:
+        super().__init__()
+        self.sample_key = sample_key
+        self.prediction_key = prediction_key
+        self.img_size = img_size if isinstance(img_size, tuple) else (img_size, img_size)
+        self.block_size = self._calcualte_block_size(self.img_size, patch_size, patch_stride, add_cls_token)
+
+        self.embedding_fn = ImagePatchEmbedding(n_img_channels, n_embd, patch_size, patch_stride, add_cls_token)
+        self.positional_embedding_fn = nn.Embedding(num_embeddings=self.block_size, embedding_dim=n_embd)
+        self.dropout = nn.Dropout(dropout)
+        self.blocks = nn.ModuleList(
+            [
+                VisionTransformerBlock(
+                    attention_config=attention_config,
+                    n_embd=n_embd,
+                    n_head=n_head,
+                    ffn_hidden=ffn_hidden,
+                    bias=bias,
+                    dropout=dropout,
+                )
+                for _ in range(n_layer)
+            ]
+        )
+
+        self.head = None
+        if n_classes is not None:
+            self.norm = nn.LayerNorm(n_embd)
+            self.head = nn.Linear(in_features=n_embd, out_features=n_classes, bias=bias)
+
+    def forward_images(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.embedding_fn(x)
+        x = self.dropout(x + self.positional_embedding_fn.weight)
+        for block in self.blocks:
+            x = block(x)
+        return x
+
+    def forward(self, inputs: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor]:
+        x = inputs[self.sample_key]
+        x = self.forward_images(x)
+        if self.head:
+            if self.embedding_fn.cls_token is not None:
+                x = x[:, 0]
+            else:
+                x = x.mean(dim=1)
+            x = self.head(self.norm(x))
+        return {self.prediction_key: x}
+
+    @staticmethod
+    def _calcualte_block_size(img_size, patch_size, patch_stride, add_cls_token):

```

Comment:
Typo in "calculate" and missing type hints. Also, would be nice to have a comment on where this formula is coming from, e.g. by adding a link.